### PR TITLE
VAULT-12833 Update prompts for the rekey command

### PR DESCRIPTION
**Pull request by miagilepner**
All the prompts for the rekey command currently assume that the `-target` flag is set to its default value of `barrier`. If the `-target=recovery` option is set, then the prompts should instead describe recovery keys. I updated the language and added some tests around recovery keys.

### CODE REVIEW

File: command/command_test.go [hunk 1]
```diff
@@ -89,13 +90,24 @@ func testVaultServerAllBackends(tb testing.TB) (*api.Client, func()) {
 	return client, closer
 }
 
+// testVaultServerAutoUnseal creates a test vault cluster and sets it up with auto unseal
+// the function returns a client, the recovery keys, and a closer function
+func testVaultServerAutoUnseal(tb testing.TB) (*api.Client, []string, func()) {
+	testSeal := seal.NewTestSeal(nil)
+	autoSeal, err := vault.NewAutoSeal(testSeal)
```
Comment (miagilepner) [inline command/command_test.go:97]: I'm not sure if there's a better way to set up auto-unseal for testing

File: command/command_test.go [hunk 2]
```diff
@@ -159,14 +173,24 @@ func testVaultServerCoreConfigWithOpts(tb testing.TB, coreConfig *vault.CoreConf
...
-		unsealKeys[i] = base64.StdEncoding.EncodeToString(cluster.BarrierKeys[i])
+	var keys [][]byte
+	if coreConfig.Seal != nil && coreConfig.Seal.RecoveryKeySupported() {
+		keys = cluster.RecoveryKeys
+	} else {
+		keys = cluster.BarrierKeys
 	}
 
-	return client, unsealKeys, func() { defer cluster.Cleanup() }
+	return client, encodeKeys(keys), func() { defer cluster.Cleanup() }
```
Comment (ncabatoff) [inline command/command_test.go:183]: I don't think `defer` makes sense here.  It won't harm, but it's not needed and it's confusing.

File: command/operator_rekey.go [hunk 3]
```diff
@@ -58,6 +63,11 @@ Usage: vault operator rekey [options] [KEY]
   the command. If key is specified as "-", the command will read from stdin. If
   a TTY is available, the command will prompt for text.
 
+  If the flag -target=recovery is supplied, then this operation will
```
Comment (ncabatoff) [inline command/operator_rekey.go:66]: You might want to add something to the effect that: in this mode, recovery keys are also used in place of the unseal keys as the inputs.  You've clarified this for the outputs, but it's left ambiguous as to the inputs.

File: command/operator_rekey.go [hunk 4]
```diff
@@ -58,6 +63,11 @@ Usage: vault operator rekey [options] [KEY]
...
+  generate a new set of recovery keys rather than unseal keys. These recovery
+  keys provide an authorization mechanism for actions that require a quorum. The
+  recovery keys will not be able to unseal vault.
```
Comment (ncabatoff) [inline command/operator_rekey.go:69]: This is actually changing in 1.13 (but only on oss), so I'd leave this out.

File: command/operator_rekey.go [hunk 5]
```diff
@@ -708,33 +721,56 @@ func (c *OperatorRekeyCommand) printUnsealKeys(client *api.Client, status *api.R
...
+				"Vault rekeyed with %d key shares and a key threshold of %d. Please "+
+					"securely distribute the key shares printed above. When Vault is "+
+					"re-sealed, restarted, or stopped, you must supply at least %d of "+
+					"these keys to unseal it before it can start servicing requests.",
+				status.N,
+				status.T,
+				status.T)))
+		case "recovery", "hsm":
+			c.UI.Output(wrapAtLength(fmt.Sprintf(
+				"Vault rekeyed with %d key shares and a key threshold of %d. Please "+
```
Comment (ncabatoff) [inline command/operator_rekey.go:736]: Can we get the target into the text here and above, e.g. "Vault %s keys rekeyed with %d..." where %s is "unseal" or "recovery"?

File: command/operator_rekey.go [hunk 6]
```diff
@@ -708,33 +721,56 @@ func (c *OperatorRekeyCommand) printUnsealKeys(client *api.Client, status *api.R
...
+					"Please securely distribute the key shares printed above. When Vault "+
+					"is re-sealed, restarted, or stopped, you must supply at least %d of "+
+					"these keys to unseal it before it can start servicing requests.",
+				status.N,
+				status.T,
+				status.T)))
+			warningText = "unseal"
+		case "recovery", "hsm":
+			c.UI.Output(wrapAtLength(fmt.Sprintf(
+				"Vault has created a new key, split into %d key shares and a key threshold "+
```
Comment (ncabatoff) [inline command/operator_rekey.go:759]: Same here, e.g. "Vault has created a new recovery key," or "a new unseal key".

File: command/command_test.go (hunk 1 above)
Comment (ncabatoff) [reply command/command_test.go:97]: Depends on what you're trying to do.  I think you made the right choice here.  If we were actually exercising seal related more directly, I might point you at something that doesn't use a NewTestSeal, which is kind of a fake.

**Review by ncabatoff — APPROVED** [review]
Looks good.  I've made some minor improvement suggestions, take 'em or leave 'em as you see fit.

File: command/operator_rekey.go (hunk 5 above)
Comment (miagilepner) [reply command/operator_rekey.go]: added

File: command/operator_rekey.go (hunk 4 above)
Comment (miagilepner) [reply command/operator_rekey.go]: Removed

**Pull request merged**